packages/exchange/src/pairs/XDaiBridge.ts

```diff
@@ -1,39 +1,126 @@
+      },
+      {
+        "indexed": false,
+        "name": "value",
+        "type": "uint256"
+      },
+      {
+        "indexed": false,
+        "name": "transactionHash",
+        "type": "bytes32"
+      }
+    ],
+    "name": "AffirmationCompleted",
+    "type": "event"
   }
+]
 
-  exchangeAtoB({ account, value, ether }: ExchangeParams) {
-    const _value = this._getValue({ value, ether });
-    const xdai = this.getExchange().getAsset('xdai');
-    return xdai.send({
-      from: account,
-      value: _value,
-      to: toDaiBridge,
-    });
+const bridgeBAbi: AbiItem[] = [
+  {
+    "anonymous": false,
+    "inputs": [
+      {
+        "indexed": false,
+        "name": "recipient",
+        "type": "address"
+      },
+      {
+        "indexed": false,
+        "name": "value",
+        "type": "uint256"
+      },
+      {
+        "indexed": false,
+        "name": "transactionHash",
+        "type": "bytes32"
+      }
+    ],
+    "name": "RelayedMessage",
+    "type": "event"
   }
+]
 
-  exchangeBtoA({ account, value, ether }: ExchangeParams) {
-    const _value = this._getValue({ value, ether });
+const wait = (time: number) => new Promise(resolve => setTimeout(resolve, time));
 
-    const dai = this.getExchange().getAsset(this.assetB);
-    return dai.send({
-      from: account,
-      value: _value,
-      to: toXdaiBridge,
+export default class XDaiBridge extends Bridge {
+  constructor() {
+    super({
+      assetA: 'xdai',
+      assetABridge: '0x7301cfa0e1756b71869e93d4e4dca5c7d0eb0aa6',
+      assetB: 'dai',
+      assetBBridge: '0x4aa42145Aa6Ebf72e164C9bBC74fbD3788045016'
     });
   }
 
-  async estimateAtoB(value: ValueTypes) {
-    return this._getValue(value);
+  async detectExchangeBToAFinished(account: string, value: string, sendResult: any) {
+    const asset = this.getExchange().getAsset(this.assetA);
+    const web3 = asset.getWeb3();
+    const contract = new web3.eth.Contract(bridgeAAbi, this.assetABridge);
+    let fromBlock = await web3.eth.getBlockNumber()
+
+    while (true) {
+      const currentBlock = await web3.eth.getBlockNumber()
+
+      const events: EventData[] = await contract.getPastEvents('AffirmationCompleted', {
+        fromBlock,
+        toBlock: currentBlock
+      })
+
+      const confirmationEvent = events.filter(event => event.returnValues.transactionHash === sendResult.txHash)
+
+      if (confirmationEvent.length > 0) {
+        const event = confirmationEvent[0]
+        const block = await web3.eth.getBlock(event.blockNumber);
+
+        const historyEvent = {
+          id: `${event.transactionHash}-${event.logIndex}`,
+          asset: asset.id,
+          type: 'send',
+          value: event.returnValues.value.toString(),
+          from: this.assetABridge,
+          to: event.returnValues.recipient,
+          tx: event.transactionHash,
+          timestamp: block.timestamp
+        }
+        // @ts-ignore
+        asset.core.addHistoryEvent(historyEvent)
```

Comment:
@dmihal  Is there a better way to access the `core` component? While it is available from the `Asset` class, the types do not mention it. That's why I had to add the `// @ts-ignore` statement.

The `xdai` asset does not support to display the native transfer from the block reward contract, that's why when a user exchanges `dai` to `xdai`, in the `Recent activity` element the incoming `xdai` to the account is not displayed. 

So the idea was to add a new event row so it is displayed in the `Recent activity` element. For example:
![changeDaiToXdai](https://user-images.githubusercontent.com/4614574/79151772-6f602500-7da1-11ea-9154-5c47ef7a3df6.png)

One issue is that when you click in the new row, in the console the error `Invalid UTF-8 detected` is thrown from the line https://github.com/austintgriffith/burner-core/blob/b24b2137a0e4d7517848b6ba9972ea835841354f/assets/src/NativeAsset.js#L157 and so `Transaction not found...` is displayed.
![receivedNotFound](https://user-images.githubusercontent.com/4614574/79151795-7edf6e00-7da1-11ea-9555-d3e48ca96ead.png)

Reply:
As explained in the other comments, I removed this logic since it better fits in the xdai `NativeAsset` methods.

Reply:
> As explained in the other comments, I removed this logic since it better fits in the xdai `NativeAsset` methods.

Could you attach a link to the corresponding PR in order to not lose relationship?

Reply:
Here is the PR https://github.com/austintgriffith/burner-core/pull/4